core/new-gui/src/app/workspace/service/workflow-result/workflow-result.service.ts

```diff
@@ -57,6 +67,51 @@ export class WorkflowResultService {
     return this.operatorResultServices.get(operatorID);
   }
 
+  private handleCleanResultCache(event: WorkflowAvailableResultEvent): void {
+    const removedOrInvalidatedOperators = new Set<string>();
+    // remove operators that no longer have results
+    this.operatorResultServices.forEach((_, op) => {
+      if (!(op in event.availableOperators)) {
+        this.operatorResultServices.delete(op);
+        removedOrInvalidatedOperators.add(op);
+      }
+    });
+    this.paginatedResultServices.forEach((_, op) => {
+      if (!(op in event.availableOperators)) {
+        this.paginatedResultServices.delete(op);
+        removedOrInvalidatedOperators.add(op);
+      }
+    });
+    // for each operator that has results:
+    Object.entries(event.availableOperators).forEach((e) => {
```

Comment:
what is this `e`? can we be more clear on the naming?

Reply:
fixed